rxmongo/src/main/scala/akka/contrib/persistence/mongodb/RxMongoReadJournaller.scala

```diff
@@ -142,11 +144,40 @@ class EventsByPersistenceId(val driver:RxMongoDriver,persistenceId:String,fromSe
   }
 }
 
+class RxMongoJournalStream(driver: RxMongoDriver) extends JournalStream[Event, Enumerator[BSONDocument]] with JournalEventBus{
+  import concurrent.ExecutionContext.Implicits.global
+  import RxMongoSerializers._
+  override def cursor = driver.realtime.find(BSONDocument.empty).options(QueryOpts().tailable.awaitData).cursor[BSONDocument]().enumerate()
+
+  private val flatten: Enumeratee[BSONDocument,Event] = Enumeratee.mapFlatten[BSONDocument] { doc =>
+    Enumerator(
+      doc.as[BSONArray](EVENTS).values.collect {
+        case d:BSONDocument => driver.deserializeJournal(d)
+      } : _*
+    )
+  }
+
+  override def streaming = {
+    publishEvent(publish)
+  }
+
+  override def publishEvent(handler: (Event) => Unit) = {
+    val iteratee = Iteratee.foreach[Event](handler)
+    cursor.through(flatten).run(iteratee)
+  }
+}
+
 class RxMongoReadJournaller(driver: RxMongoDriver) extends MongoPersistenceReadJournallingApi {
-  override def allPersistenceIds: Props = AllPersistenceIds.props(driver)
+  override def currentAllEvents: Props = CurrentAllEvents.props(driver)
 
-  override def allEvents: Props = AllEvents.props(driver)
+  override def currentPersistenceIds: Props = CurrentAllPersistenceIds.props(driver)
 
-  override def eventsByPersistenceId(persistenceId: String, fromSeq: Long, toSeq: Long): Props =
-    EventsByPersistenceId.props(driver,persistenceId,fromSeq,toSeq)
+  override def currentEventsByPersistenceId(persistenceId: String, fromSeq: Long, toSeq: Long): Props =
+    CurrentEventsByPersistenceId.props(driver,persistenceId,fromSeq,toSeq)
+
+  override def publishJournalEvents(persistenceId: String, subscriber: ActorRef) = {
+    val journalStream = new RxMongoJournalStream(driver)
```

Comment:
Should the `journalStream` be a singleton for the driver that is re-used?  Same applies to casbah